apps/react/demo/src/ui/Showcase/common/Example/hooks/useProviderState.ts

```diff
@@ -0,0 +1,113 @@
+import { ORIGINAL_VAR_NAME_KEY } from "data/index.js";
+import { useGlobalForm } from "hooks/index.js";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { useFormContext } from "react-hook-form";
+import type { ExampleOutputFormat, Output } from "../types.js";
+import type { UseProviderStateProps, UseProviderStateResult } from "./types.js";
+
+/**
+ * Hook to manage the state of the provider
+ */
+const useProviderState = ({
+  outputFormats = ["css"],
+}: UseProviderStateProps): UseProviderStateResult => {
+  // Default to the first item if available
+  const [activeExampleIndex, setActiveExampleIndex] = useState(0);
+  const { defaultValues, examples } = useGlobalForm();
+  const { setValue, getValues, watch } = useFormContext();
+
+  useEffect(() => {
+    // Currently, changes to simple-changes and select fields don't cause re-renders unless we do this.
+    watch();
+  }, [watch]);
```

Comment:
@advl This feels like a workaround or not fully the right way to do this, let me know what you think

I wanted to avoid using `formState = useWatch()` in the useEffect below as I don't actually consume `formState` - I instead use formContext.getValues, in order to drill into the nested global form state created by using period-delimited control keys.